src/main/java/com/facebook/openwifirrm/modules/ConfigManager.java

```diff
@@ -256,10 +284,10 @@ private void runImpl() {
 			logger.trace("Config changes are disabled.");
 		} else if (devicesNeedingUpdate.isEmpty()) {
 			logger.debug("No device configs to send.");
-		} else if (params.configOnEventOnly && !isEvent) {
+		} else if (params.configOnEventOnly && !shouldUpdate) {
 			// shouldn't happen
 			logger.error(
-				"ERROR!! {} device(s) queued for config update, but event flag not set",
+				"ERROR!! {} device(s) queued for config update, but set of zones to update is empty.",
```

Comment:
"set of zones to update is empty" -> "no zones queued for update" (try to keep log messages plain english)